### ISSUE TITLE
Input thread blocking causes mouse handling stutters on windows

### ISSUE THREAD

**Issue by peppy** [ppy/osu-framework#6181]
Many reports of this recently, to the point I believe something may have regressed.

https://github.com/ppy/osu/issues/27208
https://github.com/ppy/osu/discussions/26998
https://github.com/ppy/osu/discussions/26953
https://github.com/ppy/osu/discussions/27211
https://github.com/ppy/osu/discussions/26875 (maybe)

**Comment by owengaming001** [comment]
Hi, I made the https://github.com/ppy/osu/issues/27208 issue. I have since found out that this is related to my laptop keyboard, and isn't dependent on the touchpad. I've had typing lag-related issues in other programs, so it's possible that issue might not be entirely osu!lazer's fault. Though it's still worth noting that it doesn't seem to happen on stable. Also, it affected me for a while before I got around to reporting it.

Basically what I'm saying is: given the circumstances, my issue might be unrelated from the rest of those listed here. Hopefully this is helpful, sorry for any confusion or problems I may have caused. If there are any questions I can answer, then don't hesitate to let me know.

**Comment by BenCheung0422** [comment]
I believe the four issues above are all related to tablet pen and the last one is related to mouse. I think it is generally cursor handling problem.

**Comment by peppy** [comment]
Probably keep this open until a fix has been confirmed.

**Comment by Sewlell** [comment]
Creator of [ppy/osu#27448](https://github.com/ppy/osu/issues/27488) here, the new April 12 Update seem to lessened the frequency of the stutter (included other rulesets, ie. sentakki) . Unfortunately it's still unresolved with several time it can be very noticeable and caused a FC choke.


Not sure about other people though ( need more confirmation ). But hey, at least I got 50/50 chances this time for me.

**Comment by smoogipoo** [comment]
The update has not changed anything yet. This will likely be fixed or significantly improved in the next release.

**Comment by IIIBlueberry** [comment]
The issue is still here, I did not notice mouse input(Logitech G102 Prodigy 1000hz polling) dropping on my old core 2 quad 9300 and r7 360 compared to my new better rig.

**Comment by peppy** [comment]
> The issue is still here, I did not notice mouse input(Logitech G102 Prodigy 1000hz polling) dropping on my old core 2 quad 9300 and r7 360 compared to my new better rig.

To reiterate, this will only happen when something else on your system is handling input badly. For the time being, you can resolve this locally by trial-and-error shutting down apps (especially things in your system tray) until you find the culprit.

When you do, please report back as to which app it is so we can share knowlege with other users.

**Comment by IIIBlueberry** [comment]
> > The issue is still here, I did not notice mouse input(Logitech G102 Prodigy 1000hz polling) dropping on my old core 2 quad 9300 and r7 360 compared to my new better rig.
> 
> To reiterate, this will only happen when something else on your system is handling input badly. For the time being, you can resolve this locally by trial-and-error shutting down apps (especially things in your system tray) until you find the culprit.
> 
> When you do, please report back as to which app it is so we can share knowlege with other users.

I pin point it down to a few thing, 1. The input dropping happened when my Comodo firewall periodically using my NVMe drive; I started having this issues not only on osu!lazer but Aimlabs around after I'm cloned my SATA SSD to NVMe boot drive and forgot to disconnect the old SSD before booting to the new drive causing bug and conflict to Comodo Firewall making it constantly accessing drive. Disconnecting my SATA SSD and uninstalling Comodo Firewall seems to solved the input dropping for both games.

But I'm unsure if the issue either caused by CF, drive usage or my drive cloning, I suggest people task manager and check if the input dropping is synced with drive active time.

2. I noticed on my old PC(Different OS) That leaving Discord running in the background consistently causing input dropping, I'm not sure if it was because of being CPU bounded, but please try closing Discord.

**Comment by BenCheung0422** [comment]
In fact, any program, or driver, or process, could cause the issue, given that the issue is that the input thread of the game is easily affected by the system's threads. However, I always have dozens or hundreds of processes running on my computer, so this is quite an unresolvable issue on my end before the solution is launched.
